File: rust/template.rs
use std::error::Error;
use std::time::Instant;

use aoc_shared::{read_input, report_times};

fn main() -> Result<(), Box<dyn Error>> {
    let start = Instant::now();

    // read and parse file
    let input = read_input("2024", "1", true)?;

    for line in input {
        let line = line?;
        if line.is_empty() {
            continue;
        }
    }
    let file_read_time = start.elapsed();

    // part 1
    let part1_result: isize = 0;
    let part1_time = start.elapsed();

    // part 2
    let part2_result: isize = 0;
    let part2_time = start.elapsed();

    // report results and times
    println!("{part1_result}");
    println!("{part2_result}");
    report_times(file_read_time, part1_time, part2_time);
    Ok(())
}

File: c++/template.cpp
#include <fstream>
#include <iostream>

#include "aoc_shared.h"

using namespace std;
using namespace std::chrono;

auto main() -> int {
    const auto start = steady_clock::now();

    // read and parse file
    ifstream file = read_input("2024", "1", true);
    string line;
    while (getline(file, line)) {
        if (line.empty()) continue;
    }
    const auto file_read_time = steady_clock::now() - start;

    // part 1
    auto part1_result = 0;
    const auto part1_time = steady_clock::now() - start;

    // part 2
    auto part2_result = 0;
    const auto part2_time = steady_clock::now() - start;

    // report results and times
    cout << part1_result << endl;
    cout << part2_result << endl;
    report_times(file_read_time, part1_time, part2_time);
}
